docs/config.md

```diff
@@ -577,30 +730,66 @@ It will run the model `vgg_19_graphdef` over combinations of batch sizes `[1,2,3
 
 It will also run the model `vgg_16_graphdef` over combinations of batch sizes `[4,5,6,7,8,9]`(taken from the global `batch_sizes` section), concurrency `[2,10,18,26,34,42,50,58]`, with dynamic batching enabled and preferred batch sizes `1` and `2` and a single GPU instance.
 
-### `<triton-server-flags>`
+## The `analysis_models` fields and `<analysis-model>`
+The `--analysis-models` argument can be provided as a list of strings (names of models) from the CLI interface, or as a more complex `<analysis-model>` object but only through the YAML configuration file. The model object can contain `<objectives>` and `<constraints>`. An example looks like:
 
-This section of the config allows fine-grained control over the flags passed to the Triton instances launched by Model Analyzer when it is running in the `docker` or `local` Triton launch modes. Any argument to the server can be specified here.
+```yaml
+analysis_models:
+  model_1:  
+    constraints:
+        gpu_used_memory:
+            max: 200
+    objectives:
+    - perf_throughput
+```
 
-#### Example
+Multiple models can be specified under the `analysis_models` key as well.
 
 ```yaml
-model_repository: /path/to/model/repository/
-model_names:
-  model_1:
-    parameters:
-        batch_sizes:
-            start: 4
-            stop: 9
-        concurrency:
-            - 2
-            - 4
-            - 8
-triton_server_flags:
-    strict-model-config: False
-    log-verbose: True
+analysis_models:
+  model_1:  
+    constraints:
+        gpu_used_memory:
+            max: 200
+  model_1:  
+    constraints:
+        perf_latency:
+            max: 80
+objectives:
+- perf_throughput
 ```
 
-**Important Notes**: 
-1. The section name contains underscores `triton_server_flags` as seen in the example above. However the arguments themselves should contain hyphens as seen with `strict-model-config`.
-2. The Model Analyzer also provides certain arguments to the `tritonserver` instances it launches. These ***cannot*** be overriden by providing those arguments in this section. An example of this is `http-port`, which is an argument to Model Analyzer itself.
+## The `report_model_configs` fields and `<report-model-config>`
+The `--report-model-configs` argument can be provided as a list of strings (names of models) from the CLI interface, or as a more complex `<report-model-config>` object but only through the YAML configuration file. The model object can contain `<plots>` objects. An example looks like:
+
+```yaml
+report_model_configs:
+  model_1_i0:
+    throughput_v_latency:
+      title: Title
+      x_axis: perf_latency
+      y_axis: perf_throughput
+      monotonic: True
+```
+
+Multiple models can be specified under the `analysis_models` key as well.
+
+```yaml
+report_model_configs:
+  model_1_i0:
+    throughput_v_latency:
```

Comment:
Do we want to expose this API? I'm not sure whether we have properly tested all the various combinations of this API.

Reply:
Yes i think the detailed report requires that the user be able to specify the plots they want. However, I forgot to unskip the config unit test. I will fix that in a separate PR im working on because I broke one other unittest.